pkg/generate/code/initialize_field.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package code
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws-controllers-k8s/code-generator/pkg/fieldpath"
+
+	"github.com/aws-controllers-k8s/code-generator/pkg/model"
+)
+
+// InitializeNestedStructField returns the go code for initializing a nested
+// struct field. Currently this method only supports the struct shape for
+// nested elements.
+//
+// TODO(vijtrip2): Refactor the code out of set_resource.go for generating
+// constructors and reuse here. This method is currently being used for handling
+// nested Tagging fields.
+//
+// Example: generated code for "Logging.LoggingEnabled.TargetBucket" field
+// inside "s3" "bucket" crd looks like:
+//
+//  ```
+// 	r.ko.Spec.Logging = &svcapitypes.BucketLoggingStatus{}
+//	r.ko.Spec.Logging.LoggingEnabled = &svcapitypes.LoggingEnabled{}
+//  ```
+func InitializeNestedStructField(
+	r *model.CRD,
+	sourceVarName string,
+	field *model.Field,
+	// apiPkgAlias contains the imported package alias where the type definition
+	// for nested structs is present.
+	// ex: svcapitypes "github.com/aws-controllers-k8s/s3-controller/apis/v1alpha1"
+	apiPkgAlias string,
+	// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	out := ""
+	indent := strings.Repeat("\t", indentLevel)
+	fieldPath := field.Path
+	if fieldPath != "" {
+		fp := fieldpath.FromString(fieldPath)
+		if fp.Size() > 1 {
```

Comment:
nit: Early exit on inverse (`if fp.Size() == 0`)

Reply:
not needed, this is the only condition. if `fp.Size() == 0`, it never enters if block and jumps to return statement.